ludwig/features/category_feature.py

```diff
@@ -59,7 +59,11 @@ class _CategoryPreprocessing(torch.nn.Module):
     def __init__(self, metadata: Dict[str, Any]):
         super().__init__()
         self.str2idx = metadata["str2idx"]
-        self.unk = self.str2idx[UNKNOWN_SYMBOL]
+        if UNKNOWN_SYMBOL in self.str2idx:
+            self.unk = self.str2idx[UNKNOWN_SYMBOL]
+        else:
+            # TODO: Replace with mode or something to that effect
+            self.unk = 0
```

Comment:
I can replace this with the same thing I'm doing in lines 153-158, but want to make sure that this approach even makes sense to begin with and it is the cleanest way to do things @justinxzhao

Reply:
Another potential alternative to try would be to set `self.unk` as None, and add a conditional in `forward()`. This should be able to get around torchscript tracing (maybe?)

```python
if self.unk:
    indices = [self.str2idx.get(s.strip(), self.unk) for s in v]
else:
    indices = [self.str2idx[s.strip()] for s in v]
return torch.tensor(indices, dtype=torch.int32)
```

CC: @geoffreyangus

Reply:
FYI I tried this and torchscript complained. My guess is that torchscript tracing can't handle condition jumps and needs a more clear data flow to infer things.